ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -42,85 +38,39 @@ type RayDashboardClientInterface interface {
 	DeleteJob(ctx context.Context, jobName string) error
 }
 
-type BaseDashboardClient struct {
-	client       *http.Client
-	dashboardURL string
-}
-
-func GetRayDashboardClientFunc(mgr ctrl.Manager, useKubernetesProxy bool) func() RayDashboardClientInterface {
+func GetRayDashboardClientFunc(host string, useKubernetesProxy bool) func() RayDashboardClientInterface {
 	return func() RayDashboardClientInterface {
 		return &RayDashboardClient{
-			mgr:                mgr,
+			host:               host,
 			useKubernetesProxy: useKubernetesProxy,
+			client:             &http.Client{},
+			dashboardURL:       "",
 		}
 	}
 }
 
 type RayDashboardClient struct {
-	mgr ctrl.Manager
-	BaseDashboardClient
+	host               string
+	client             *http.Client
+	dashboardURL       string
 	useKubernetesProxy bool
 }
 
 // FetchHeadServiceURL fetches the URL that consists of the FQDN for the RayCluster's head service
 // and the port with the given port name (defaultPortName).
-func FetchHeadServiceURL(ctx context.Context, cli client.Client, rayCluster *rayv1.RayCluster, defaultPortName string) (string, error) {
-	log := ctrl.LoggerFrom(ctx)
-	headSvc := &corev1.Service{}
-	headSvcName, err := GenerateHeadServiceName(RayClusterCRD, rayCluster.Spec, rayCluster.Name)
-	if err != nil {
-		log.Error(err, "Failed to generate head service name", "RayCluster name", rayCluster.Name, "RayCluster spec", rayCluster.Spec)
-		return "", err
-	}
-
-	if err = cli.Get(ctx, client.ObjectKey{Name: headSvcName, Namespace: rayCluster.Namespace}, headSvc); err != nil {
-		if errors.IsNotFound(err) {
-			log.Error(err, "Head service is not found", "head service name", headSvcName, "namespace", rayCluster.Namespace)
-		}
-		return "", err
-	}
-
-	log.Info("FetchHeadServiceURL", "head service name", headSvc.Name, "namespace", headSvc.Namespace)
-	servicePorts := headSvc.Spec.Ports
-	port := int32(-1)
-
-	for _, servicePort := range servicePorts {
-		if servicePort.Name == defaultPortName {
-			port = servicePort.Port
-			break
-		}
-	}
-
-	if port == int32(-1) {
-		return "", fmtErrors.Errorf("%s port is not found", defaultPortName)
-	}
-
-	domainName := GetClusterDomainName()
-	headServiceURL := fmt.Sprintf("%s.%s.svc.%s:%v",
-		headSvc.Name,
-		headSvc.Namespace,
-		domainName,
-		port)
-	log.Info("FetchHeadServiceURL", "head service URL", headServiceURL)
-	return headServiceURL, nil
-}
-
-func (r *RayDashboardClient) InitClient(ctx context.Context, url string, rayCluster *rayv1.RayCluster) error {
-	log := ctrl.LoggerFrom(ctx)
 
+func (r *RayDashboardClient) InitClient(url string, rayCluster *rayv1.RayCluster, client *http.Client) error {
```

Comment:
1. Remove `host` from `type RayDashboardClient struct`
2. Remove `useKubernetesProxy` from `type RayDashboardClient struct`

```suggestion
func (r *RayDashboardClient) InitClient(dashboardURL string, client *http.Client) error {
```